storage/remote/read_test.go

```diff
@@ -93,7 +93,7 @@ func TestNoDuplicateReadConfigs(t *testing.T) {
 	}
 
 	for _, tc := range cases {
-		s := NewStorage(nil, prometheus.DefaultRegisterer, nil, dir, defaultFlushDeadline)
```

Comment:
was this change required for something?

Reply:
yeah, actually. `NewStorage` always calls `NewWriteStorage` even if there aren't any remote write configs, so the metrics were getting re-registered for subsequent iterations of the loop here

Reply:
Ah right I see now. This is another place you could just pass nil for the registerer if we don't care about any metric values during the test.